posthog-react-native/src/hooks/useFeatureFlag.ts

```diff
@@ -18,3 +19,26 @@ export function useFeatureFlag(flag: string): string | boolean | undefined {
 
   return featureFlag
 }
+
+export function useFeatureFlagWithPayload(flag: string): [string | boolean | undefined, JsonType | undefined]   {
+  const posthog = usePostHog()
+
+  const [featureFlag, setFeatureFlag] = useState<boolean | string | undefined>(posthog?.getFeatureFlag(flag))
+  const [featureFlagPayload, setFeatureFlagPayload] = useState<JsonType | undefined>(posthog?.getFeatureFlag(flag))
```

Comment:
Not sure I love that these two can get out of sync. 99% of the time this shouldn't happen, but still. I think we should have one state object for both values like:
```suggestion
  const [featureflagState, setFeatureFlagState] = useState<[string | boolean | undefined, JsonType | undefined]>
```

meaning later we can just do 
```
setFeatureFlag([
  posthog.getFeatureFlag(flag)),
  posthog.getFeatureFlagPayload(flag))
])
```